lib/core-integration/src/Test/Integration/Scenario/API/Shelley/StakePools.hs

```diff
@@ -215,12 +217,32 @@ spec = describe "SHELLEY_STAKE_POOLS" $ do
         pool:_ <- map (view #id) . snd <$>
             unsafeRequest @[ApiStakePool] ctx
             (Link.listStakePools arbitraryStake) Empty
-        joinStakePool @n ctx pool (src, fixturePassphrase) >>= flip verify
+        rJoin <- joinStakePool @n ctx pool (src, fixturePassphrase)
+        verify rJoin
             [ expectResponseCode HTTP.status202
             , expectField (#status . #getApiT) (`shouldBe` Pending)
             , expectField (#direction . #getApiT) (`shouldBe` Outgoing)
+            -- all tx inputs have address and amount
+            , expectField #inputs $ \inputs' -> do
+                let addrAmt = (fmap source inputs')
+                addrAmt `shouldSatisfy` (all isJust)
```

Comment:
@KtorZ as far as your simplifications to the checks, I see you have removed this one and also the one from multi-output transactions. I'd argue that these are redundant, especially that one above was failing before (not because of `fixtureWallet` thing). It is OK now (after https://github.com/input-output-hk/cardano-wallet/pull/2370 I suppose) but still worth checking.

Reply:
Whoops, sorry, you didn't remove that. It was the commit diff that mislead me :sweat_smile:
Anyway... I have added small additional check to multi-output tx in cdf556a.